boot/boot.go

```diff
@@ -403,12 +405,79 @@ func makeBootable20(model *asserts.Model, rootdir string, bootWith *BootableSet,
 	return nil
 }
 
+// XXX: kept this here for now until we have a decision of the file names and locations,
+// then we refactor to integrate this functionality to existing infrastructure. (Currently
+// bootloader.ExtractKernelAssets tests for meta/force-kernel-extraction and
+// bootloader.extractKernelAssetsToBootDir extracts to a subdir)
+func extractKernelAssets(dstDir string, s snap.PlaceInfo, snapf snap.Container) error {
+	dir, err := os.Open(dstDir)
+	if err != nil {
+		return err
+	}
+	defer dir.Close()
+
+	for _, src := range []string{"kernel.img", "initrd.img"} {
```

Comment:
we want to only extract `kernel.efi`, it it exists.

`kernel.img/initrd.img` is fallback for non-efi / non-uefi cases.

Reply:
kernel.img/initrd.img are gone in my latest kernel snap, and submitted to the kernel team to drop from edge.